### PR TITLE
Allow wifi interfaces configuration without requiring a network_key,

### PR DESCRIPTION
**Pull request by buzzdeee**
i.e. allow unencrypted wifi network configuration.

Update specs to reflect that change, add a few more wifi interfaces
I own and reorder them alphabetically. Add a little test.

Travis fails, due to :lint test in Rakefile,
removing that :lint test, tests are passing.

Don't know why it is failing, but puppet-lint also fails for me on
my local box with the same error on your master branch, without
me doing any changes :(

### CODE REVIEW

**Comment by xaque208** [comment]
I'm sorry I missed this.  Reviewing the travis failure now.

**Comment by xaque208** [comment]
I've no real idea about those failures.  Do the tests run locally with a `bundle install && bundle exec rake test` ?  Maybe a rebase is in order.

**Comment by xaque208** [comment]
Tests have been a bit improved in https://github.com/xaque208/puppet-bsd/pull/37

**Comment by buzzdeee** [comment]
btw, tested running the tests locally with bundle exec rake test, and figured, that the syntax check test was/is fine, same as the specs test, however, the second test, the lint is where it bails out.

File: tests/openbsd_wifi.pp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+bsd::network::interface::wifi { 'wpi0':
+network_name => 'myssid',
+address      => [ 'dhcp', ],
```
Comment (xaque208) [inline tests/openbsd_wifi.pp]: Two things I think are causing test failure.  Indent line 2 and 3 and remove the trailing comma inside the array.

**Comment by xaque208** [comment]
Once the issue above is fixed and the test passes, I'll merge.

**Comment by buzzdeee** [comment]
with the test indentet, the linter doesn't seem to fail anymore, and tests managed to be greeen again., or one of your other changes did that.

**Comment by xaque208** [comment]
Excellent, thanks for the PR.

**Pull request merged**